### PR TITLE
Syn & quote dependencies upgrade

### LINKED ISSUE #53: Update dependencies
`packed_struct_codegen` depends on a really old version of `syn`, `v0.11.11` from April 2017. This causes syn to be compiled multiple times, as a new version for serde and an old version for `packed_struct_codegen` are both compiled. It would be cool if `packed_struct` would be modernized a bit in general. This very issue also resulted in the [removal of `packed_struct` as dependency from cranelift](https://github.com/bytecodealliance/cranelift/issues/1271).

### PR DESCRIPTION
**Pull request by rudib**
Upgraded to `syn` 1.0 and `quote` 1.0, integrated with `proc_macro2` for testability. Added span-based error messages.

Closes #53

### CODE REVIEW

File: packed_struct_codegen/Cargo.toml [hunk 1]
```diff
@@ -7,13 +7,11 @@ license = "MIT OR Apache-2.0"
 authors = ["Rudi Benkovic <rudi.benkovic@gmail.com>"]
 
 [dependencies]
-quote = "^0.3.12"
+quote = "1.0"
```
Comment (wucke13) [inline packed_struct_codegen/Cargo.toml:10]: Consider using the caret here, as described in the [docs](https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#caret-requirements)? Same holds true for syn and proc-macro2?

Comment (rudib) [reply packed_struct_codegen/Cargo.toml:10]: The paragraph just above the linked docs says that if no operator is present then the caret is implied, so this should be effectively equivalent? It resolves to `quote v1.0.7` in the CI build right now, so it is working :)

Comment (wucke13) [reply packed_struct_codegen/Cargo.toml:10]: There we have it, now I should have done more research :)

**Comment by wucke13** [comment]
It's release time :)

**Comment by rudib** [comment]
@wucke13  0.4.0 has been released. Thanks for pinging me about this lib, it gave me the much-needed motivation to put together a new release :)

**Comment by wucke13** [comment]
I'm glad it happened as well :+1:

**Pull request merged**